receiver/receivercreator/observerhandler.go

```diff
@@ -67,9 +67,18 @@ func (obs *observerHandler) OnAdd(added []observer.Endpoint) {
 
 	for _, e := range added {
 		for _, template := range obs.receiverTemplates {
-			if !ruleMatches(template.Rule, e) {
+			if matches, err := template.rule.eval(e); err != nil {
+				obs.logger.Error("failed matching rule", zap.String("rule", template.Rule), zap.Error(err))
+				continue
+			} else if !matches {
 				continue
 			}
+
+			obs.logger.Info("starting receiver",
```

Comment:
Should this be `Debug` log instead? I am not sure how often this happens. Is this typically hapening only when new pods are added? That is probably fine, it is not too often.

Reply:
Only when a pod is added that matches a rule and a receiver is started in response.

Reply:
OK, I think it is infrequent enough. Should be fine to log it.